### PR TITLE
dumpling: Expose option for skipping PK sort

### LINKED ISSUE #31097: Dumpling: The `SortByPk` option is missing a flag
### 1. Minimal reproduce step (Required)

Inspect the code: There is a `SortByPk` option, but no flag to set it.

This causes issues when exporting data from Vitess as that doesn't support queries like this if the table is sharded:
```sql
SELECT * FROM tbl ORDER BY col1;
```

However it does support:
```sql
SELECT col1, col2, ... FROM tbl ORDER BY col1;
```

Or:
```sql
SELECT * FROM tbl
```

The column in the `ORDER BY` has to be explicitly named in the `SELECT`-list.

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?
Issue Number: close #31097 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below) Check the config dump when running `./dumpling --sort-by-pk=false`
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
An option was added to skip primary key sorting.
```

### CODE REVIEW

**Comment by sunzhaoyang** [comment]
this command line arguments LGTM

**Review by tangenta — APPROVED** [review]

File: dumpling/export/config.go [hunk 1]
```diff
@@ -226,6 +227,7 @@ func (conf *Config) DefineFlags(flags *pflag.FlagSet) {
 	flags.String(flagConsistency, consistencyTypeAuto, "Consistency level during dumping: {auto|none|flush|lock|snapshot}")
 	flags.String(flagSnapshot, "", "Snapshot position (uint64 or MySQL style string timestamp). Valid only when consistency=snapshot")
 	flags.BoolP(flagNoViews, "W", true, "Do not dump views")
+	flags.BoolP(flagSortByPk, "", true, "Sort by PK")
```
Comment (lichunzhu) [inline dumpling/export/config.go:230]:
```suggestion
	flags.Bool(flagSortByPk, true, "Sort dump results by primary key through order by sql")
```

File: dumpling/export/config.go [hunk 2]
```diff
@@ -45,6 +45,7 @@ const (
 	flagConsistency              = "consistency"
 	flagSnapshot                 = "snapshot"
 	flagNoViews                  = "no-views"
+	flagSortByPk                 = "sort-by-pk"
```
Comment (lichunzhu) [inline dumpling/export/config.go:48]:
```suggestion
	flagSortByPk                 = "order-by-primary-key"
```

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Pull request merged**